File: No.2490-Bees-master/api/src/main/java/org/bees/biz/service/BannerService.java
package org.bees.biz.service;

import java.lang.reflect.InvocationTargetException;
import java.util.LinkedList;
import java.util.List;

import org.apache.commons.beanutils.PropertyUtils;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.bees.biz.persistence.manager.BannerManager;
import org.bees.biz.persistence.model.Banner;
import org.bees.biz.service.dto.BannerDto;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class BannerService {
	
	@Autowired
	private BannerManager bannerManager;
	
	
	@SuppressWarnings("unused")
	private final static Log LOG = LogFactory.getLog(BannerService.class);

	public Banner getInfoById(Integer id) {		
		return this.bannerManager.getInfoById(id);
	}

	public List<BannerDto> getList(String page, String position, Integer start, Integer limit) {
		List<Banner> Banners=this.bannerManager.getList(page,position,start,limit);
		List<BannerDto> dtos=new LinkedList<BannerDto>();
		for (Banner banner : Banners) {
			BannerDto dto=new BannerDto();
			try {
				PropertyUtils.copyProperties(dto, banner);
				dto.setUrl(banner.getPicHref());
				dto.setImgUrl(banner.getPicUrl());
			} catch (IllegalAccessException | InvocationTargetException
					| NoSuchMethodException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
			dtos.add(dto);
		}		
		return dtos;
	}

	public boolean add(Integer userId, String title, String content,
			String page, String position, int picOrder, String picHref,
			String picUrl) {		
		return this.bannerManager.add(userId,title,content,page,position,picOrder,picHref,picUrl);
	}

	public boolean del(Integer id) {		
		return this.bannerManager.del(id);
	}

	public boolean edit(Integer id, String title, String content, String page,
			String position, int picOrder, String picHref, String picUrl) {
		return this.bannerManager.edit(id,title,content,page,position,picOrder,picHref,picUrl);
	}

	public int getListCount(String page, String position) {
		return this.bannerManager.getListCount(page, position);
	}

	
	
}

File: beesInterface/bees_banner/bannerService.py
# conding=utf-8
from models import Banner
from utils.validateUtils import ValidateUtils
from utils.const import Const

class BannerService:
	@staticmethod
	def getInfoById(id):
		banner = None
		try:
			banner = Banner.objects.get(id=id, del_flag=Const.DEL_FLAG_NOT_DEL)
		except:
			return None
		return banner

	@staticmethod
	def getList(page, position, start, limit):
		banners = Banner.objects.filter(del_flag=Const.DEL_FLAG_NOT_DEL).order_by('-id')
		if not ValidateUtils.isNull(page):
			banners = banners.filter(page=page)
		if not ValidateUtils.isNull(position):
			banners = banners.filter(position=position)
		if ValidateUtils.isNull(start):
			start = 0
		else:
			start = int(start)
		if ValidateUtils.isNull(limit):
			banners = banners[start:]
		else:
			limit = int(limit)
			banners = banners[start:start+limit]
		dtos = []
		for banner in banners:
			dtos.append(banner.getBannerDto())
		return dtos

	@staticmethod
	def add(userId, title, content, page, position, picOrder, picHref, picUrl):
		banner = Banner()
		banner.id = userId
		banner.title = title
		banner.content = content
		banner.page = page
		banner.position = position
		banner.pic_order = picOrder
		banner.pic_href = picHref
		banner.pic_url = picUrl
		banner.save()
		return True

	@staticmethod
	def delete(id):
		banner = None
		try:
			banner = Banner.objects.get(id=id)
		except:
			return False
		banner.delete()
		return True

	@staticmethod
	def edit(id, title, content, page, position, picOrder, picHref, picUrl):
		banner = None
		try:
			banner = Banner.objects.get(id=id)
		except:
			return False
		banner.title = title
		banner.content = content
		banner.page = page
		banner.position = position
		banner.pic_order = picOrder
		banner.pic_href = picHref
		banner.pic_url = picUrl
		banner.save()
		return True

	@staticmethod
	def getListCount(page, position):
		return Banner.objects.filter(page=page, position=position).count()